src/udx.c

```diff
@@ -1562,6 +1551,9 @@ on_uv_poll (uv_poll_t *handle, int status, int events) {
   }
 
   if (events & UV_WRITABLE) {
+    if (read) {
+      uv_update_time(handle->loop);
```

Comment:
time is updated on each loop tick and where we do it manually also?

Reply:
ya makes sense. i think you can just do that here by checking if events & UV_READABLE instead of the extra bool